backend/app/src/main/java/com/moh/phlat/backend/service/DbUtilityServiceImpl.java

```diff
@@ -129,31 +138,142 @@ public void setControlStatus(Long controlId, String statusCode, String authentic
+				columnnDisplayNames.add("HDS Type");
 				isValid = false;
 				logger.info("Required check failed on process data id: {}", processData.getId());
 				Message msg = Message.builder()
-									 .messageType("ERROR")
-									 .messageCode("101")
-									 .messageDesc("HDS Name cannot be empty")
-									 .sourceSystemName(MessageSourceSystem.PLR)
+						             .messageType(ProcessDataService.PHLAT_ERROR_TYPE)
+						             .messageCode(ProcessDataService.PHLAT_ERROR_CODE)
+						             .messageDesc(messageSource.getMessage(PROCESS_DATA_MANDATORY_COLUMN_DISPLAY_NAME,
+								            columnnDisplayNames.toArray() , LocaleContextHolder.getLocale()))
+									 .sourceSystemName(MessageSourceSystem.PHLAT)
 									 .processData(processData)
 									 .build();
 				messageService.createMessage(msg);
+			}
 
+			// mandatory check for PCN Srvc Delivery Type
+			if (!StringUtils.hasText(processData.getPcnServiceDeliveryType()) && processData.getStakeholder().equals("PAS")) {
+				columnnDisplayNames.clear();
+				columnnDisplayNames.add("PCN Srvc Delivery Type");
+				isValid = false;
+				logger.info("Required check failed on process data id: {}", processData.getId());
+				Message msg = Message.builder()
+						             .messageType(ProcessDataService.PHLAT_ERROR_TYPE)
+						             .messageCode(ProcessDataService.PHLAT_ERROR_CODE)
+						             .messageDesc(messageSource.getMessage(PROCESS_DATA_MANDATORY_COLUMN_DISPLAY_NAME,
+								            columnnDisplayNames.toArray() , LocaleContextHolder.getLocale()))
+									 .sourceSystemName(MessageSourceSystem.PHLAT)
+									 .processData(processData)
+									 .build();
+				messageService.createMessage(msg);
 			}
-			// error detection
-			
-			
-			if (isValid) { 
-				setProcessDataStatus(processData.getId(), RowStatusService.VALID,authenticatedUserId);
-			} else {
-				setProcessDataStatus(processData.getId(), RowStatusService.INVALID, authenticatedUserId);
+
+			// mandatory check for PCN Clinic  Type
+			if (!StringUtils.hasText(processData.getPcnClinicType()) && processData.getStakeholder().equals("PAS")) {
+				columnnDisplayNames.clear();
+				columnnDisplayNames.add("PCN Clinic Type");
+				isValid = false;
+				logger.info("Required check failed on process data id: {}", processData.getId());
+				Message msg = Message.builder()
+						             .messageType(ProcessDataService.PHLAT_ERROR_TYPE)
+						             .messageCode(ProcessDataService.PHLAT_ERROR_CODE)
+						             .messageDesc(messageSource.getMessage(PROCESS_DATA_MANDATORY_COLUMN_DISPLAY_NAME,
+								            columnnDisplayNames.toArray() , LocaleContextHolder.getLocale()))
+									 .sourceSystemName(MessageSourceSystem.PHLAT)
+									 .processData(processData)
+									 .build();
+				messageService.createMessage(msg);
 			}
+
+		}
+
 			
+		if (isValid) { 
+			setProcessDataStatus(processData.getId(), RowStatusService.VALID,authenticatedUserId);
```

Comment:
this call will not be required. Saving ProcessData entity that is retrieved, will save error message and any other attributes of process data object.